### ISSUE TITLE
4.1.0 deb not compatible with Ubuntu 20.0x and variants

### ISSUE THREAD

**Issue by celentanos** [arsenetar/dupeguru#763]
**Bug description**
Installed dupeguru version: **dupeguru_4.1.0_amd64.deb**
After dupeguru start in the terminal shows following error message:
        
        Failed to load module "xapp-gtk3-module"
        Traceback (most recent call last):
          File "/usr/bin/dupeguru", line 88, in <module>
            sys.exit(main())
          File "/usr/bin/dupeguru", line 71, in main
            from qt.app import DupeGuru
          File "/usr/share/dupeguru/qt/app.py", line 22, in <module>
            from core.app import AppMode, DupeGuru as DupeGuruModel
          File "/usr/share/dupeguru/core/app.py", line 24, in <module>
            from . import se, me, pe
          File "/usr/share/dupeguru/core/pe/__init__.py", line 1, in <module>
            from . import (  # noqa
          File "/usr/share/dupeguru/core/pe/block.py", line 9, in <module>
            from ._block import NoBlocksError, DifferentBlockCountError, avgdiff, getblocks2  # NOQA
        ModuleNotFoundError: No module named 'core.pe._block'
        Error in sys.excepthook:
        Traceback (most recent call last):
          File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 153, in apport_excepthook
            with os.fdopen(os.open(pr_filename,
        FileNotFoundError: [Errno 2] Datei oder Verzeichnis nicht gefunden: '/var/crash/_usr_share_dupeguru_run.py.1000.crash'
        
        Original exception was:
        Traceback (most recent call last):
          File "/usr/bin/dupeguru", line 88, in <module>
            sys.exit(main())
          File "/usr/bin/dupeguru", line 71, in main
            from qt.app import DupeGuru
          File "/usr/share/dupeguru/qt/app.py", line 22, in <module>
            from core.app import AppMode, DupeGuru as DupeGuruModel
          File "/usr/share/dupeguru/core/app.py", line 24, in <module>
            from . import se, me, pe
          File "/usr/share/dupeguru/core/pe/__init__.py", line 1, in <module>
            from . import (  # noqa
          File "/usr/share/dupeguru/core/pe/block.py", line 9, in <module>
            from ._block import NoBlocksError, DifferentBlockCountError, avgdiff, getblocks2  # NOQA
        ModuleNotFoundError: No module named 'core.pe._block'

**Desktop**
 - OS: Linux Mint 20.01 Xfce 64-bit

**Comment by qarmin** [comment]
Similar error I have on Ubuntu 20.04
```
Traceback (most recent call last):
  File "/usr/bin/dupeguru", line 88, in <module>
    sys.exit(main())
  File "/usr/bin/dupeguru", line 71, in main
    from qt.app import DupeGuru
  File "/usr/share/dupeguru/qt/app.py", line 22, in <module>
    from core.app import AppMode, DupeGuru as DupeGuruModel
  File "/usr/share/dupeguru/core/app.py", line 24, in <module>
    from . import se, me, pe
  File "/usr/share/dupeguru/core/pe/__init__.py", line 1, in <module>
    from . import (  # noqa
  File "/usr/share/dupeguru/core/pe/block.py", line 9, in <module>
    from ._block import NoBlocksError, DifferentBlockCountError, avgdiff, getblocks2  # NOQA
ModuleNotFoundError: No module named 'core.pe._block'
```

**Comment by arsenetar** [comment]
This is due to the deb being built on/for Debian 10.7 which uses python 3.7 and the compiled modules are built against 3.7.  Ubuntu 20.x (20.01, 20.04) and derivatives ship with python 3.8 which the compiled modules are not compatible with.  Options right now are build from source following the instructions in the ReadMe or build from the debian source archive.

**Comment by qarmin** [comment]
It is really strange.
I was think always that python was designed to work on different OS with backward compatibility (3.7 should be not so different than 3.8 and most of things should work in newer versions without change)

**Comment by arsenetar** [comment]
@qarmin The particular modules causing the issue are compiled python extensions the rest of the files would be just fine between versions (baring changes like between 2.x and 3.x).  The modules need to have a particular filename for python to load it (in the case that there are differences between the versions).  I think these modules should be able to run between 3.7 and 3.8 without issue so one could rename or create links to the _block modules and the _cache module .so files to load them under python 3.8 (file names currently are marked for python 3.7 with `37m` which needs to be `38` to load in python 3.8).

**Comment by arsenetar** [comment]
Also just want to note although it seems in this particular case renaming/symlinking does appear to work, it is not guaranteed. So to work across all python versions supported would need to include versions of the three modules for 3.6, 3.7, 3.8, & 3.9.

**Comment by xplosionmind** [comment]
I downloaded dupeGuru 4.1.0 on my Ubuntu 20.10 workstation, I click on the icon to open the program and nothing happens. Literallly nothing. Is this related to this issue?

**Comment by qarmin** [comment]
Yes.
You should see error when opening DupeGuru from terminal.

**Comment by FlorentCoppint** [comment]
Hi,
Here is 4.1.0 build for Ubuntu 20.04 & 20.10 (Python 3.8) : https://1fichier.com/?vnefwm65mob7nc039k0p
VirusTotal result : https://www.virustotal.com/gui/url/ba2b49459824d7394f55c25c0b9e6ec2078a99a62f5aae01a58a038200b0766d/detection

**Comment by Mannshoch** [comment]
any news about a release?

**Comment by arsenetar** [comment]
Fixed in release 4.1.1
